### ISSUE TITLE
Wedos DNS challenge fails

### ISSUE THREAD

**Issue by typekcz** [go-acme/lego#1840]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest releases.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### What did you expect to see?

Successfully generated certificate.

### What did you see instead?

Could not obtain certificates:
```
[example.com] [example.com] acme: error presenting token: wedos: no subdomain because the domain and the zone are identical: example.com.
```

### How do you use lego?

Docker image

### Reproduction steps

_I have replaced my domain with example.com in all snippets and outputs._
```
docker run --env WEDOS_USERNAME="xxx@xxx" --env WEDOS_WAPI_PASSWORD="xxx" goacme/lego --email="xxx@xx" --domains="test.example.com" --dns wedos --accept-tos run
```

### Version of lego

```console
lego version v4.10.0 linux/amd64
```

### Logs

<details>

```console
2023/02/17 13:09:53 No key found for account xxx@xxx. Generating a P256 key.
2023/02/17 13:09:53 Saved key to /.lego/accounts/acme-v02.api.letsencrypt.org/xxx@xxx/keys/xxx@xxx.key
2023/02/17 13:09:54 [INFO] acme: Registering account for xxx@xxx
2023/02/17 13:09:54 [INFO] [test.example.com] acme: Obtaining bundled SAN certificate
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2023/02/17 13:09:54 [INFO] [test.example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/204201146476
2023/02/17 13:09:54 [INFO] [test.example.com] acme: Could not find solver for: tls-alpn-01
2023/02/17 13:09:54 [INFO] [test.example.com] acme: Could not find solver for: http-01
2023/02/17 13:09:54 [INFO] [test.example.com] acme: use dns-01 solver
2023/02/17 13:09:54 [INFO] [test.example.com] acme: Preparing to solve DNS-01
2023/02/17 13:09:54 [INFO] Found CNAME entry for "_acme-challenge.test.example.com.": "example.com."
2023/02/17 13:10:04 [INFO] [test.example.com] acme: Cleaning DNS-01 challenge
2023/02/17 13:10:04 [INFO] Found CNAME entry for "_acme-challenge.test.example.com.": "example.com."
2023/02/17 13:10:14 [WARN] [test.example.com] acme: cleaning up failed: wedos: no subdomain because the domain and the zone are identical: example.com.
2023/02/17 13:10:14 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/204201146476
2023/02/17 13:10:15 Could not obtain certificates:
        error: one or more domains had a problem:
[test.example.com] [test.example.com] acme: error presenting token: wedos: no subdomain because the domain and the zone are identical: example.com.
```

</details>

**Comment by ldez** [comment]
Hello,

are you using a CNAME?

**Comment by typekcz** [comment]
Hello,

thanks for the quick response. I am not using CNAME.
Only records I have are:
```
example.com A
example.com AAAA
*.example.com A
*.example.com AAAA
```

**Comment by ldez** [comment]
:thinking: maybe it's related to the wildcard A record.

can you try to set the env var `LEGO_DISABLE_CNAME_SUPPORT` to `true`?

**Comment by typekcz** [comment]
That is working. I had to fiddle around with few more things, but that's not part of this issue.
So from my side, it is closed, since I don't need CNAME support.
Thank you.

**Comment by rien** [comment]
I'm having the same issue with `hetzner` now. @typekcz what did you do to fix this issue?

**Comment by typekcz** [comment]
So I was dealing with another issue after I have set `LEGO_DISABLE_CNAME_SUPPORT` to `true`.
I don't remember what was the exact issue, but my local DNS server had different DNS records that the internet ones.
That was intentional, but I didn't know that lego was checking created TXT record by querying my default DNS server.
So I added this parameter: `--dns.resolvers 1.1.1.1 8.8.8.8` and then everything was fine.
Thinking back maybe that was also causing the error that I described in this issue.
